src/ffi.rs

```diff
@@ -213,8 +214,12 @@ impl CoordSeq {
         CoordSeq(unsafe { GEOSCoordSeq_create(size as c_uint, dims as c_uint) })
     }
 
-    pub fn new_from_c_obj(c_obj: *mut GEOSCoordSequence) -> CoordSeq {
-        CoordSeq(c_obj)
+    pub fn new_from_c_obj(c_obj: *mut GEOSCoordSequence) -> GeosResult<CoordSeq> {
+        if c_obj.is_null() {
+            Err(Error::NoConstructionFromNullPtr)
+        } else {
+            Ok(CoordSeq(c_obj))
+        }
     }
     pub fn set_x(&self, idx: u32, val: f64) -> i32 {
```

Comment:
Given that it may also fail, should not these 3 methods (`set_x`, `set_y` and  `set_z`) also return a `Result`? (rather than letting the user look at the return value)

Reply:
oh right, I missed those, thanks !